components/titiler/mapbox-map.js

```diff
@@ -0,0 +1,87 @@
+import React, {useRef, useEffect, useState} from 'react';
+import mapboxgl from 'mapbox-gl'
+import { Box, Container, useThemeUI } from 'theme-ui'
+import axios from 'axios'
+import getConfig from 'next/config'
+
+const querystring = require('querystring');
+const { publicRuntimeConfig } = getConfig()
+
+mapboxgl.accessToken = publicRuntimeConfig.MAPBOX_TOKEN
+const XARRAY_TILER_ENDPOINT = 'https://dev-titiler-xarray.delta-backend.com/tilejson.json'
+const API_STAC_ENDPOINT = 'https://staging-stac.delta-backend.com/'
+const stacCollectionID = 'oco2-geos-l3-daily'
+
+const tileData = {
+    resampling_method:'bilinear',
+    variable:'XCO2',
+    colormap_name:'rdylbu_r',
+    rescale:'0.00039394,0.000420'
+}
+
+export default function SimpleMap() {
+  const mapRef = useRef(null);
+  const [zarrTileParams, setZarrTileParams] = useState('')
+  const [mapStyle, setMapStyle] = useState(undefined)
+  const [baseStyle, setBaseStyle] = useState(undefined);
+  
+  useEffect(() => {
+    if (!mapRef.current) return
+    const mbMap = new mapboxgl.Map({
+      container: mapRef.current,
+      attributionControl: false,
+      style: 'mapbox://styles/mapbox/light-v11',
+      zoom: 4
+    });
+    mapRef.current = mbMap;
+    const controller = new AbortController();
+    async function getTileParams() {
+        const stacRes = await axios.get(`${API_STAC_ENDPOINT}/collections/${stacCollectionID}`, {
+            signal: controller.signal
+         });
+        const zarrAssetUrl = stacRes.data.assets.zarr.href;
```

Comment:
In case this helps, this is the response data.
```
{
    "id": "oco2-geos-l3-daily",
    "type": "Collection",
    "links": [
        {
            "rel": "items",
            "type": "application/geo+json",
            "href": "https://staging-stac.delta-backend.com/collections/oco2-geos-l3-daily/items"
        },
        {
            "rel": "parent",
            "type": "application/json",
            "href": "https://staging-stac.delta-backend.com/"
        },
        {
            "rel": "root",
            "type": "application/json",
            "href": "https://staging-stac.delta-backend.com/"
        },
        {
            "rel": "self",
            "type": "application/json",
            "href": "https://staging-stac.delta-backend.com/collections/oco2-geos-l3-daily"
        },
        {
            "rel": "external",
            "href": "https://catalog.data.gov/dataset/oco-2-geos-level-3-daily-0-5x0-625-assimilated-co2-v10r-oco2-geos-l3co2-day-at-ges-disc-72b15",
            "type": "text/html",
            "title": "OCO-2 GEOS Level 3 daily, 0.5x0.625 assimilated CO2 V10r (OCO2_GEOS_L3CO2_DAY) at GES DISC",
            "label:assets": null
        }
    ],
    "title": "Gridded Daily OCO-2 Carbon Dioxide assimilated dataset",
    "assets": {
        "zarr": {
            "href": "s3://veda-data-store-staging/EIS/zarr/OCO2_GEOS_L3CO2_day.zarr",
            "type": "application/vnd+zarr",
            "roles": [
                "data",
                "zarr"
            ],
            "title": "Zarr Array Store",
            "description": "Zarr array store with one or several arrays (variables)",
            "xarray:open_kwargs": {
                "chunks": {},
                "engine": "zarr",
                "consolidated": true
            }
        }
    },
    "extent": {
        "spatial": {
            "bbox": [
                [
                    -180,
                    -90,
                    180,
                    90
                ]
            ]
        },
        "temporal": {
            "interval": [
                [
                    "2015-01-01T12:00:00+00:00",
                    "2021-11-04T12:00:00+00:00"
                ]
            ]
        }
    },
    "license": "CC0-1.0",
    "keywords": null,
    "providers": null,
    "summaries": null,
    "description": "The OCO-2 mission provides the highest quality space-based XCO2 retrievals to date. However, the instrument data are characterized by large gaps in coverage due to OCO-2’s narrow 10-km ground track and an inability to see through clouds and thick aerosols. This global gridded dataset is produced using a data assimilation technique commonly referred to as state estimation within the geophysical literature. Data assimilation synthesizes simulations and observations, adjusting the state of atmospheric constituents like CO2 to reflect observed values, thus gap-filling observations when and where they are unavailable based on previous observations and short transport simulations by GEOS. Compared to other methods, data assimilation has the advantage that it makes estimates based on our collective scientific understanding, notably of the Earth's carbon cycle and atmospheric transport. OCO-2 GEOS (Goddard Earth Observing System) Level 3 data are produced by ingesting OCO-2 L2 retrievals every 6 hours with GEOS CoDAS, a modeling and data assimilation system maintained by NASA's Global Modeling and Assimilation Office (GMAO). GEOS CoDAS uses a high-performance computing implementation of the Gridpoint Statistical Interpolation approach for solving the state estimation problem. GSI finds the analyzed state that minimizes the three-dimensional variational (3D-Var) cost function formulation of the state estimation problem.",
    "item_assets": null,
    "stac_version": "1.0.0",
    "cube:variables": {
        "XCO2": {
            "type": "data",
            "unit": "mol CO2/mol dry",
            "attrs": {
                "units": "mol CO2/mol dry",
                "long_name": "Assimilated dry-air column average CO2 daily mean"
            },
            "shape": [
                2500,
                361,
                576
            ],
            "chunks": [
                100,
                100,
                100
            ],
            "dimensions": [
                "time",
                "lat",
                "lon"
            ],
            "description": "Assimilated dry-air column average CO2 daily mean"
        },
        "XCO2PREC": {
            "type": "data",
            "unit": "mol CO2/mol dry",
            "attrs": {
                "units": "mol CO2/mol dry",
                "long_name": "Precision of dry-air column average CO2 daily mean from Desroziers et al. (2005) diagnostic"
            },
            "shape": [
                2500,
                361,
                576
            ],
            "chunks": [
                100,
                100,
                100
            ],
            "dimensions": [
                "time",
                "lat",
                "lon"
            ],
            "description": "Precision of dry-air column average CO2 daily mean from Desroziers et al. (2005) diagnostic"
        }
    },
    "cube:dimensions": {
        "lat": {
            "axis": "y",
            "type": "spatial",
            "extent": [
                -90,
                90
            ],
            "description": "latitude",
            "reference_system": 4326
        },
        "lon": {
            "axis": "x",
            "type": "spatial",
            "extent": [
                -180,
                179.375
            ],
            "description": "longitude",
            "reference_system": 4326
        },
        "time": {
            "step": "P1DT0H0M0S",
            "type": "temporal",
            "extent": [
                "2015-01-01T12:00:00Z",
                "2021-11-04T12:00:00Z"
            ],
            "description": "time"
        }
    },
    "stac_extensions": [
        "https://stac-extensions.github.io/datacube/v2.0.0/schema.json"
    ],
    "dashboard:is_periodic": true,
    "dashboard:time_density": "day"
}
```

Reply:
I was misunderstanding the task that this prototype should embrace the dynamic data from stac endpoint. closing.